### ISSUE TITLE
Research BRMS fingerprints

### ISSUE THREAD

**Issue by noahl** [quipucords/rho#484]
As the title says.

**Comment by noahl** [comment]
What I know so far:

BRMS is supported on a *lot* of containers: JBoss EAP, Apache Tomcat, JBoss Fuse (I don't think this is a container, but I'm listing it because the BRMS docs do too), JBoss Web Server, IBM WebSphere, and Oracle WebLogic (https://access.redhat.com/documentation/en-us/red_hat_jboss_brms/6.4/html/installation_guide/chap_introduction). I'm asking for information on which are the most common. Testing on all of the above could be very difficult.

Luckily, it looks like BRMS uses the same jars on all systems. I think an approach that makes sense is
  1. Use locate to find the jars, wherever they might be on the system
  2. In addition, on EAP systems, look in the known EAP directories to find jars

Once we have the jars, we're set regardless.

Adding more containers would mean finding ways to detect other container systems and then finding what directories we should look for jars in with those systems. Adding more containers would only help us in the case that a user was running BRMS on a non-EAP container and didn't have locate installed.

**Comment by noahl** [comment]
I have now looked inside all (as far as I know) distributions of BRMS. A BRMS installations has two key folders, business-central and kie-server. In EAP installations, these folders have `.war` on the end. In the generic form, they do not. In an EAP installation, they appear at `$EAP_HOME/standalone/deployments`. In a Tomcat installation, they will be at `tomcat7/webapps`.

These two folders indicate that BRMS is present. If we find candidate business-central and kie-server folders, we can verify that they are the right thing by looking at `business-central/META-INF/MANIFEST.MF` and `kie-server/META-INF/MANIFEST.MF`. (This is vulnerable to tampering, but see below.)

We will look for business-central and kie-server in two ways: using the locate command, and if we have a known `$EAP_HOME` directory, by looking at the relative paths I listed above.

Looking at `business-central(.war)/WEB-INF/lib/kie-api-****.jar` gives the BRMS version. This also adds some resiliency to tampering, since it would presumably be hard to remove this jar from an installation without breaking BRMS.
